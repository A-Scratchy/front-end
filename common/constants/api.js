import { get, post, patch } from 'common/utils/api-utils';
import { coerceEmptyStringToUndefined } from 'common/utils/string-utils';

/* GET REQUESTS */
export const getCodeSchoolsPromise = () => get('code_schools');
export const getMentorPromise = id => get(`mentors/${id}`);
export const getMentorsPromise = () => get('mentors');
export const getRequestsPromise = () => get('requests');
export const getScholarshipPromise = id => get(`scholarships/${id}`);
export const getScholarshipsPromise = () => get('scholarships');
export const getServicesPromise = () => get('services');
export const getTeamMembersPromise = () => get('team_members');

/* POST REQUESTS */
export const createUser = ({ email, password, firstName, lastName, zipcode }) =>
  post('users', {
    user: {
      email,
      password,
      first_name: firstName,
      last_name: lastName,
      zip: zipcode,
    },
  }).then(({ data }) => data);

export const loginUser = ({ email, password }) =>
  post('sessions', {
    user: {
      email,
      password,
    },
  }).then(({ data }) => data);

export const postMentorRequestPromise = ({ language, additionalDetails, mentor, service }) =>
  post('requests', {
    request: {
      details: additionalDetails,
      language,
      requested_mentor_id: mentor,
      service_id: service,
    },
  });

/* PATCH REQUESTS */
export const updateUser = ({
  branchOfService,
  companyName,
  companyRole,
  disciplines,
  doesWantMentor, // TODO: Use in form
  doesWantScholarshipInfo, // TODO: Use in form
  doesWantToVolunteer, // TODO: Use in form
  employmentStatus,
  militaryStatus,
  payGrade,
  programmingLanguages,
  yearsOfService,
}) => {
  const interestItems = [...programmingLanguages, ...disciplines];

  // Interests could be an array of many filled strings, one filled string, or one empty string
  const interests =
    interestItems.length > 1
      ? interestItems.join(', ')
      : [coerceEmptyStringToUndefined(interestItems[0])];

  return patch('users', {
    user: {
      branch_of_service: coerceEmptyStringToUndefined(branchOfService),
      company_name: coerceEmptyStringToUndefined(companyName),
      company_role: coerceEmptyStringToUndefined(companyRole),
      mentor: doesWantMentor,
      scholarship_info: doesWantScholarshipInfo,
      volunteer: doesWantToVolunteer,
      employment_status: coerceEmptyStringToUndefined(employmentStatus),
      military_status:
        militaryStatus === 'civilian' ? undefined : coerceEmptyStringToUndefined(militaryStatus),
      pay_grade: coerceEmptyStringToUndefined(payGrade),
      interests,
      years_of_service: coerceEmptyStringToUndefined(yearsOfService),
    },
  }).then(({ data }) => data);
};

export const patchUpdateMentorRequestPromise = ({ request, status, mentor }) =>
  patch(`requests/${request}`, {
    request: {
      status,
      mentor,
    },
  });
