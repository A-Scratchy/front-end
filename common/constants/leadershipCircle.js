// Instructions:

// Below is a list. To do this, you must do the following:
// 1:
// 2:
// 3:
// 4:
// 5:
// 6:
// 7:

const honorRoll = [{}];

export default honorRoll;
