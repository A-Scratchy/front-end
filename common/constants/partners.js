const partners = [
  {
    name: 'APEX Systems',
    logo: 'partnerLogos/apex_systems.png',
    url: 'https://apexsystems.com',
  },
  {
    name: 'GitHub',
    logo: 'partnerLogos/github.png',
    url: 'https://github.com',
  },
  {
    name: '1Password',
    logo: 'partnerLogos/1password.png',
    url: 'https://1password.com/',
  },
  {
    name: 'Docker',
    logo: 'partnerLogos/docker.png',
    url: 'https://docker.com',
  },
  {
    name: 'Oracle',
    logo: 'partnerLogos/oracle.png',
    url: 'https://oracle.com',
  },
  {
    name: "O'Reilly Media",
    logo: 'partnerLogos/oreilly.png',
    url: 'https://oreilly.com',
  },
  {
    name: 'HackerRank',
    logo: 'partnerLogos/hackerrank.png',
    url: 'https://hackerrank.com',
  },
  {
    name: 'Zapier',
    logo: 'partnerLogos/zapier.png',
    url: 'https://zapier.com',
  },
];

export default partners;
