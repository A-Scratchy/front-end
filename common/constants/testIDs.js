export const ALERT = 'ALERT';
export const ALERT_CLOSE_BUTTON = 'Close Alert Button';
export const BUTTON = 'BUTTON';
export const CHANGE_PASSWORD_FORM_ERROR = 'CHANGE_PASSWORD_FORM_ERROR';
export const FLAT_CARD_IMAGE = 'FLAT_CARD_IMAGE';
export const INPUT = 'INPUT';
export const INPUT_ERROR = 'INPUT_ERROR';
export const INPUT_FEEDBACK_GROUPING = 'INPUT_FEEDBACK_GROUPING';
export const LABEL = 'LABEL';
export const SCREEN_READER_ONLY = 'SCREEN_READER_ONLY';
