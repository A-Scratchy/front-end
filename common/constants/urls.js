/* eslint-disable max-len */
export const s3 = 'https://s3.amazonaws.com/operationcode-assets/';
export const donateLink = 'https://secure.lglforms.com/form_engine/s/BRtP7QUKyHOyEYsZROsRew';
export const leadershipCircleLink =
  'https://secure.lglforms.com/form_engine/s/L428AQ2rrsFJQyy5Fbglvg';
export const footerLinks = {
  column1: [
    {
      url: '/about',
      title: 'About Us',
    },
    {
      url: '/team',
      title: 'Team',
    },
    {
      url: '/contact',
      title: 'Contact Us',
    },
    {
      url: '/faq',
      title: 'FAQ',
    },
  ],
  column2: [
    {
      url: '/who_we_serve',
      title: 'Who We Serve',
    },
    {
      url: '/code_schools',
      title: 'Code Schools',
    },
    {
      url: '/job_board',
      title: 'Job Board',
    },
    {
      url: '/events',
      title: 'Events',
    },
    {
      url: '/blog',
      title: 'Blog',
    },
  ],
  column3: [
    {
      url: '/about',
      title: 'About Us',
    },
    {
      url: '/team',
      title: 'Team',
    },
    {
      url: '/contact',
      title: 'Contact Us',
    },
    {
      url: '/faq',
      title: 'FAQ',
    },
  ],
  column4: [
    {
      url: '/resources',
      title: 'Resources',
    },
    {
      url: '/press',
      title: 'Press',
    },
    {
      url: '/branding',
      title: 'Branding',
    },
  ],
  legal: [
    {
      analyticsEventLabel: 'FooterLink',
      url: '',
      title: 'Terms of Use',
    },
    {
      analyticsEventLabel: 'FooterLink',
      url: '',
      title: 'Cookies',
    },
    {
      analyticsEventLabel: 'FooterLink',
      url: 'https://www.iubenda.com/privacy-policy/8174861',
      title: 'Privacy',
    },
  ],
};
