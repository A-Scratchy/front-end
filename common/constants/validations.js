export const minPasswordCharNum = 8;

export const validationErrorMessages = {
  required: 'Required',
  email: 'Must be a valid email',
  emailMatch: 'Emails must match',
  password: 'Must include the following: lowercase letter, uppercase letter, number',
  passwordMatch: 'Passwords must match',
  length: val => `Must be at least ${val} characters`,
  zipcode: 'Must be a valid zipcode',
};
