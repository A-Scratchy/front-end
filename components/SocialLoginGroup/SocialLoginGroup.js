import React from 'react';
import { string, func } from 'prop-types';
import classNames from 'classnames';
import FacebookLogin from 'react-facebook-login/dist/facebook-login-render-props';
import GoogleLogin from 'react-google-login';
import { googleKey, facebookKey } from 'common/config/environment';
import { loginSocial } from 'common/constants/api';
import { getErrorMessage } from 'common/utils/api-utils';
import Alert from 'components/Alert/Alert';
import styles from './SocialLoginGroup.css';

class SocialLoginGroup extends React.Component {
  static propTypes = {
    className: string,
    handleSuccess: func.isRequired,
  };

  static defaultProps = {
    className: undefined,
  };

  state = {
    errorMessage: '',
  };

  onSuccess = provider => async ({ accessToken }) => {
    try {
      const { handleSuccess } = this.props;
      const result = await loginSocial(provider, { accessToken });
      handleSuccess(result);
    } catch (error) {
      this.setState({ errorMessage: getErrorMessage(error) });
    }
  };

  onGoogleFailure = () => {
    this.setState({ errorMessage: "Couldn't log in with Google" });
  };

  render() {
    const { errorMessage } = this.state;
    const { className } = this.props;

    return (
      <>
        <div className={classNames(className, styles.flexRow, styles.SocialLoginGroup)}>
          <Alert
            className={classNames(styles.flexRow, styles.fullWidth)}
            isOpen={Boolean(errorMessage)}
            type="error"
          >
            {errorMessage}
          </Alert>

          <GoogleLogin
            clientId={googleKey}
            buttonText="Login"
            onSuccess={this.onSuccess('google')}
            onFailure={this.onGoogleFailure}
            render={renderProps => (
              <button
                type="button"
                onClick={renderProps.onClick}
                disabled={renderProps.disabled}
                className={classNames(styles.loginButton, styles.googleButton)}
              >
                Login with Google
              </button>
            )}
          />

          <FacebookLogin
            appId={facebookKey}
            callback={this.onSuccess('facebook')}
            render={renderProps => (
              <button
                type="button"
                onClick={renderProps.onClick}
                className={classNames(styles.loginButton, styles.facebookButton)}
              >
                Login with Facebook
              </button>
            )}
          />
        </div>
      </>
    );
  }
}

export default SocialLoginGroup;
