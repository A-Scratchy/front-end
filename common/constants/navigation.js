import { donateLink } from 'common/constants/urls';

// links shared between nav and footer (no duplicate code)
const contactLink = {
  name: 'Contact Us',
  href: '/contact',
};

const faqLink = {
  name: 'FAQ',
  href: '/faq',
};

const getInvolvedLink = {
  name: 'Get Involved',
  href: '/get_involved',
};

const eventsLink = {
  name: 'Events',
  href: '/events',
};

export const whoWeServeSection = {
  name: 'Who We Serve',
  href: '/who_we_serve',
  shouldPrefetch: false,
  sublinks: [
    {
      name: 'Member Login',
      href: '/login',
    },
    {
      name: 'Join',
      href: '/join',
    },
  ],
};

const logoutLink = {
  name: 'Logout',
  href: '/login?loggedOut=true',
  shouldPrefetch: false,
  sublinks: [],
};

export const navItems = [
  {
    name: 'About Us',
    href: '/about',
    shouldPrefetch: false,
    sublinks: [contactLink, faqLink],
  },
  {
    ...eventsLink,
    shouldPrefetch: false,
    sublinks: [],
  },
  {
    ...getInvolvedLink,
    shouldPrefetch: false,
    sublinks: [
      // {
      //   name: 'Mentoring',
      //   href: '/mentoring',
      // },
      {
        name: 'Sponsorship',
        href: '/sponsorship',
      },
      {
        name: 'Leadership Circle',
        href: '/leadership_circle',
      },
    ],
  },
];

export const loggedInNavItems = [...navItems, logoutLink];
export const loggedOutNavItems = [...navItems.slice(0, 1), whoWeServeSection, ...navItems.slice(1)];

export const footerItems = {
  column1: [
    {
      href: '/about',
      name: 'About Us',
    },
    contactLink,
    faqLink,
  ],
  column2: [
    {
      href: '/code_schools',
      name: 'Code Schools',
    },
    {
      href: '/jobs',
      name: 'Job Board',
    },
    eventsLink,
    {
      href: '/who_we_serve',
      name: 'Who We Serve',
    },
  ],
  column3: [
    getInvolvedLink,
    // {
    //   href: '/mentoring',
    //   name: 'Become A Mentor',
    // },
    {
      href: '/history',
      name: 'History',
    },
    {
      href: donateLink,
      name: 'Donate',
      analyticsEventLabel: 'Donate',
    },
  ],
  column4: [
    // {
    //   href: '/resources',
    //   name: 'Resources',
    // },
    {
      href: '/press',
      name: 'Press',
    },
    {
      href: '/branding',
      name: 'Branding',
    },
    {
      href: '/team',
      name: 'Team',
    },
    // {
    //   href: '/blog',
    //   name: 'Blog',
    // },
  ],
  legal: [
    {
      href: '/terms',
      name: 'Terms of Use',
    },
    // TODO: Create a link for this
    // {
    //   href: 'https://www.anotherfakeurl.com/',
    //   name: 'Cookies',
    //   analyticsEventLabel: 'Cookies',
    // },
    {
      // NOTE: If you change this route, please update the redirect in `now.json` as well
      href: 'https://www.iubenda.com/privacy-policy/8174861',
      name: 'Privacy',
      analyticsEventLabel: 'Privacy',
    },
  ],
};
