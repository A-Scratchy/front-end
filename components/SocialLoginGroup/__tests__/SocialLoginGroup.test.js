/* eslint-env jest */
import React from 'react';
import createSnapshotTest from 'test-utils/createSnapshotTest';
import { mount } from 'enzyme';
import asyncRenderDiff from 'test-utils/asyncRenderDiff';
import wait from 'test-utils/wait';
import { loginSocial } from 'common/constants/api';
import OperationCodeAPIMock from 'test-utils/mocks/apiMock';
import SocialLoginGroup from '../SocialLoginGroup';
import SocialLoginButtons from '../SocialLoginButtons';

const socialReturnToken = { accessToken: 'abc123' };

beforeEach(() => {
  OperationCodeAPIMock.reset();
});

function renderWithHelpers() {
  const handleSuccessSpy = jest.fn();

  let renderProps;

  const wrapper = mount(
    <SocialLoginGroup
      className="test-class"
      loginSocial={loginSocial}
      handleSuccess={handleSuccessSpy}
    >
      {({ onSuccess, onGoogleFailure }) => {
        renderProps = { onSuccess, onGoogleFailure };
      }}
    </SocialLoginGroup>,
  );
  return { wrapper, handleSuccessSpy, renderProps };
}

describe('SocialLoginGroup', () => {
  it('should render with required props', () => {
    createSnapshotTest(
      <SocialLoginGroup className="test-class" loginSocial={jest.fn()} handleSuccess={jest.fn()}>
        {() => {}}
      </SocialLoginGroup>,
    );
  });

  it('should render SocialLoginButtons with required props', () => {
    createSnapshotTest(<SocialLoginButtons onGoogleFailure={jest.fn()} onSuccess={jest.fn()} />);
  });

  it('calls callbacks when onSuccess is triggered', async () => {
    OperationCodeAPIMock.onPost('auth/social/google/', socialReturnToken).reply(200);

    const { wrapper, handleSuccessSpy, renderProps } = renderWithHelpers();

    const onSuccess = renderProps.onSuccess('google');
    await onSuccess(socialReturnToken);

    await asyncRenderDiff(wrapper);

    await wait(() => {
      expect(handleSuccessSpy).toHaveBeenCalledWith('google', socialReturnToken);
    });
  });

  it('does NOT call handleSuccess when loginSocial fails', async () => {
    const { wrapper, handleSuccessSpy, renderProps } = renderWithHelpers();

    OperationCodeAPIMock.onPost('auth/social/google/', socialReturnToken).reply(400, {
      error: 'User is already registered with this e-mail address.',
    });

    const onSuccess = renderProps.onSuccess('google');
    await onSuccess(socialReturnToken);

    await asyncRenderDiff(wrapper);

    await wait(() => {
      expect(handleSuccessSpy).not.toHaveBeenCalled();
    });

    expect(wrapper.find('Alert').text()).toStrictEqual(
      'User is already registered with this e-mail address.',
    );
  });
});
