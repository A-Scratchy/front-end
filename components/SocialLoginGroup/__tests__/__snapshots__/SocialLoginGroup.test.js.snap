// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialLoginGroup should render with required props 1`] = `
<div
  className="test-class flexRow SocialLoginGroup"
>
  <div
    className="Alert flexRow fullWidth error"
    role="alert"
  >
    
  </div>
  <button
    className="loginButton googleButton"
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    Login with Google
  </button>
  <button
    className="loginButton facebookButton"
    onClick={[Function]}
    type="button"
  >
    Login with Facebook
  </button>
</div>
`;
